db/models/articles-model.js

```diff
@@ -97,6 +97,47 @@ exports.createArticleComment = (article_id, username, body) => {
         }  
     })
 }
-// exports.updateArticleById =() => {
-
-// }
\ No newline at end of file
+exports.updateArticleById =(votes_inc, article_id) => {  
+    if(isNaN(votes_inc)) {
+        return Promise.reject({
+            status: 400, message: 'invalid data type'
+        })
+    }
+    if(isNaN(article_id)) {
+        return Promise.reject({
+            status: 400, message: 'article_id must be a number'
+        })
+    }
+    return db
+    .query(
+        `SELECT article_id FROM articles
+         WHERE article_id = $1`, [article_id])
+    .then((result)=> {
+        const { rows } = result
+        if (rows.length === 0) {
+            return Promise.reject({ 
+                status: 404, message: 'article_id does not exist'
+            })
+        }
+    }
+    ).then(() => {
+    return db
+    .query(
+        `SELECT votes FROM articles
+         WHERE article_id = $1`, [article_id]
+    )
+    })
+    .then((result)=> {
+        const existingVotes = result.rows[0].votes
+        const updatedVotes = existingVotes + votes_inc
+       return db
```

Comment:
We should be chaining a .then on to this query like we have done before and sending back just the response at index [0] so that we aren't needing to implement this array logic in the controller. As a general rule all logic should be in the model.

Reply:
done